crates/wasi-nn/tests/all.rs

```diff
@@ -13,41 +13,30 @@ const PREOPENED_DIR_NAME: &str = "fixture";
 /// Run a wasi-nn test program. This is modeled after
 /// `crates/wasi/tests/all/main.rs` but still uses the older preview1 API for
 /// file reads.
-fn run(path: &str, preload_model: bool) -> Result<()> {
+fn run(path: &str, backend: Backend, preload_model: bool) -> Result<()> {
     wasmtime_wasi_nn::check_test!();
     let path = Path::new(path);
-    let config = Config::new();
-    let engine = Engine::new(&config)?;
+    let engine = Engine::new(&Config::new())?;
     let mut linker = Linker::new(&engine);
     wasmtime_wasi_nn::witx::add_to_linker(&mut linker, |s: &mut Ctx| &mut s.wasi_nn)?;
     wasi_common::sync::add_to_linker(&mut linker, |s: &mut Ctx| &mut s.wasi)?;
     let module = Module::from_file(&engine, path)?;
-    let mut backends = vec![];
-    #[cfg(feature = "openvino")]
-    {
-        backends.push(Backend::from(backend::openvino::OpenvinoBackend::default()));
-    }
-    #[cfg(feature = "winml")]
-    {
-        backends.push(Backend::from(backend::winml::WinMLBackend::default()));
-    }
-    for backend in backends {
```

Comment:
@abrown this won't work with multiple features enabled. For example, if onnx and openvino are enabled, then each backend will try to be run for a given model. Unless the backends support the same model input, the caller is going to have a bad time.

Rather than running for every backend in a test, I have each test pass in their backend to run.

Reply:
Yeah, that makes more sense! 👍